subiquity/common/tests/test_ubuntu_advantage.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from subprocess import CalledProcessError, CompletedProcess
+import unittest
+from unittest.mock import patch
+
+from subiquity.common.ubuntu_advantage import (
+    InvalidUATokenError,
+    ExpiredUATokenError,
+    CheckSubscriptionError,
+    UAInterface,
+    MockedUAInterfaceStrategy,
+    UAClientUAInterfaceStrategy,
+    )
+from subiquitycore.tests.util import run_coro
+
+
+class TestMockedUAInterfaceStrategy(unittest.TestCase):
+    def setUp(self):
+        self.strategy = MockedUAInterfaceStrategy(scale_factor=1_000_000)
+
+    def test_query_info_invalid(self):
+        # Tokens starting with "i" in dry-run mode cause the token to be
```

Comment:
are the i/f/x sentinel values valid starting token characters?
If so I'd like to pick another strategy for dryrun.

Reply:
I haven't gotten an "official" answer on this yet but I believe they would indeed be valid starting token characters.
Any suggestion for a different strategy would be welcome. Do you think picking non alphanumerical characters would be enough?

Reply:
OK.  I think the risk is low of a collision.  I accept this strategy.